src/v/kafka/requests/kafka_batch_adapter.cc

```diff
@@ -123,22 +119,41 @@ void kafka_batch_adapter::verify_crc(int32_t expected_crc, iobuf_parser in) {
     }
 }
 
-void kafka_batch_adapter::adapt(iobuf&& kbatch) {
+iobuf kafka_batch_adapter::adapt(iobuf&& kbatch) {
+    batch = std::nullopt;
+    // kafka_length_diff is because the batch size given in the kafka
+    // header does not include the offset preceeding the length field nor
+    // the size of the length field itself.
+    constexpr size_t kafka_length_diff = sizeof(int64_t) + sizeof(int32_t);
+    if (unlikely(kbatch.size_bytes() < kafka_length_diff)) {
```

Comment:
confused by this statement. 

it says 12 bytes, but below you say start at byte 4 and use 12 bytes as distance. 

So one of them to me reads like a bug. To me you either want to start at 4th byte and read 8 bytes, or start at 0 and read 12 bytes, or check for 16 bytes.

Reply:
I want to read batch_length (4bytes) after the base_offset (8 bytes)